src/ui/control/logo_control.js

```diff
@@ -21,9 +21,14 @@ class LogoControl {
     onAdd(map) {
         this._map = map;
         this._container = DOM.create('div', 'mapboxgl-ctrl');
+        const anchor = DOM.create('a', 'mapboxgl-ctrl-logo');
+        anchor.target = "_blank";
+        anchor.href = "https://www.mapbox.com/";
+        anchor.setAttribute("aria-label", "Mapbox logo");
+        this._container.appendChild(anchor);
+        this._container.style.display = 'none';
 
         this._map.on('sourcedata', this._updateLogo);
-        this._updateLogo();
```

Comment:
I think updating on add was a good idea. Instead of removing it, how about changing the condition in `_updateLogo` to `!e || e.sourceDataType === 'metadata')`?